build_tools/services.rb

```diff
@@ -9,10 +9,10 @@ class ServiceEnumerator
     MANIFEST_PATH = File.expand_path('../../services.json', __FILE__)
 
     # Minimum `aws-sdk-core` version for new gem builds
-    MINIMUM_CORE_VERSION = "3.228.0"
+    MINIMUM_CORE_VERSION = "3.231.0"
 
     # Minimum `aws-sdk-core` version for new S3 gem builds
-    MINIMUM_CORE_VERSION_S3 = "3.228.0"
+    MINIMUM_CORE_VERSION_S3 = "3.231.0"
```

Comment:
I actually think we don't need to bump s3 min core since this `credential_source` usage only applies within Amazon EC2 instances or Amazon Elastic Container Service containers